lib/modules/accountSwitcher.js

```diff
@@ -312,21 +317,24 @@ function findMatchingAccount(username, partialMatch) {
 async function switchTo(user) {
 	const account = findMatchingAccount(user);
 	if (!account) return;
-	const [username, password] = account;
+	const [username, password, uses2Fa] = account;
 
-	if (loggedInUser()) {
-		await ajax({
-			method: 'POST',
-			url: '/logout',
-		});
+	// Don't await promise yet so that the PIN can be entered while logging out
+	const logoutPromise = loggedInUser() && ajax({ method: 'POST', url: '/logout' });
+
+	let pin;
+	if (uses2Fa) {
+		pin = window.prompt(`Enter 2FA PIN for user ${user}`);
 	}
 
+	await logoutPromise;
+
 	const { success, jquery } = await ajax({
 		method: 'POST',
 		url: '/api/login',
 		data: {
 			user: username,
-			passwd: password,
+			passwd: pin ? `${password}:${pin}` : password,
```

Comment:
Alternatively, this could be more structured:
```
{
    user: username,
    passwd: password,
    otp_sent: pin,
    otp: pin,
}
```

Reply:
Setting `otp_sent` to false seemingly didn't make the server ignore the `otp` field.